### PR TITLE
[mds-audit] Allow providers to read back bulk audit records

### PR DESCRIPTION
**Pull request by avatarneil**
## 📚 Purpose
Super starter code for adding the ability for providers to read their own audits. Inspired by some of the discussions today in the City Services WG! Totally throw-awayable, but thought it would be fun to proof out.

## 📦 Impacts:
[mds-audit]

## ✅ PR Checklist
- [ ] Add or remove checklist items to suit your needs

### CODE REVIEW

**Comment by avatarneil** [comment]
> That was fast!

Hahaha, nowhere near done yet, just the beginnings of a POC :)

**Comment by avatarneil** [comment]
TODO in another PR -- add reading back for single audit

File: packages/mds-audit/api.ts [hunk 1]
```diff
@@ -624,16 +624,34 @@ function api(app: express.Express): express.Express {
...
+          'audit_subject_id'
+        )
+
+        const provider_id = scopes.includes('audits:read') ? queried_provider_id : res.locals.claims?.provider_id
+
+        if (provider_id === null) {
+          /* This should never happen -- a client with just the audits:read:provider scope
+           * should *always* have a provider_id claim in their token.
+           */
+          return res.status(500).send({ error: 'internal_server_error' })
```
Comment (janedotx) [inline packages/mds-audit/api.ts:645]: Does it say in the spec that this must return the string `internal_server_error`? Usually we create a new `ServerError` to return for 500s.

File: packages/mds-audit/tests/api.spec.ts [hunk 2]
```diff
@@ -735,6 +735,34 @@ describe('Testing API', () => {
...
+    })
+
+    it("verify provider cannot get others' trips", done => {
+      request
+        .get('/audit/trips')
+        .set('Authorization', SCOPED_AUTH(['audits:read:provider'], uuid())) // Generate arbitrary provider_id
+        .expect(200)
+        .end((err, result) => {
+          /* Expect a 200 response with no audits returned */
+          test.value(result.body.audits).is([])
```
Comment (janedotx) [inline packages/mds-audit/tests/api.spec.ts:761]: Hoory for validating the response body.

File: packages/mds-audit/api.ts (hunk 1 above)
Comment (avatarneil) [reply packages/mds-audit/api.ts:645]: I was just following the same `500` response that was in the rest of the endpoint ¯\_(ツ)_/¯

**Pull request merged**